internal/include/otel_types.h

```diff
@@ -102,13 +102,16 @@ static __always_inline void convert_go_otel_attributes(void *attrs_buf, s64 slic
 		return;
 	}
 
-	s64 num_attrs = slice_len < OTEL_ATTRUBUTE_MAX_COUNT ? slice_len : OTEL_ATTRUBUTE_MAX_COUNT;
+	u8 num_attrs = slice_len < OTEL_ATTRUBUTE_MAX_COUNT ? slice_len : OTEL_ATTRUBUTE_MAX_COUNT;
 	go_otel_key_value_t *go_attr = (go_otel_key_value_t*)attrs_buf;
 	go_otel_attr_value_t go_attr_value = {0};
 	struct go_string go_str = {0};
-	u8 valid_attrs = 0;
+	u8 valid_attrs = enc_attrs->valid_attrs;
+	if (valid_attrs >= OTEL_ATTRUBUTE_MAX_COUNT) {
+		return;
+	}
 
-	for (u32 go_attr_index = 0; go_attr_index < num_attrs; go_attr_index++) {
+	for (u8 go_attr_index = 0; go_attr_index < num_attrs; go_attr_index++) {
```

Comment:
thought: just as a general note - this was really tricky to find. without downsizing `go_attr_index` and `num_attrs`, i was getting an ebpf error where the max number of instructions was being exceeded, which with the default logs you only see that the verifier prints a general "too many arguments" error. this eventually became more clear when i reduced the noise in the logs and only enabled `ebpf.LogLevelStats`.

Reply:
Right, we should probably change the default verifier log settings.